geode-core/src/test/java/org/apache/geode/internal/cache/PartitionedRegionTest.java

```diff
@@ -637,6 +645,85 @@ public void testNotifyRegionCreated() {
     assertThat(partitionedRegion.isRegionCreateNotified()).isTrue();
   }
 
+  @Test
+  public void populateEmptyIndexesThrowsIfBucketRegionDestroyedDueToCacheClose() {
+    PartitionedRegion spyPartitionedRegion = spy(partitionedRegion);
+    BucketRegion bucketRegion = mock(BucketRegion.class);
+    when(bucketRegion.isDestroyed()).thenReturn(true);
+
+    Set<Index> indexes = setupIndexes();
+    ConcurrentMap<Integer, BucketRegion> map = setupBuckets(bucketRegion);
+    setupDataStore(spyPartitionedRegion, map);
+    CacheClosedException cacheClosedException = new CacheClosedException();
+    setupCancelCriterion(cacheClosedException);
+
+    assertThatThrownBy(() -> spyPartitionedRegion.populateEmptyIndexes(indexes, new HashMap<>()))
+        .isInstanceOf(CancelException.class).isEqualTo(cacheClosedException);
```

Comment:
I think that it's redundant to have both `.isInstanceOf(CancelException.class)` and `.isEqualTo(cacheClosedException)`, since it's not possible that the exception is equal to `cacheClosedException` but NOT an instance of `CancelException`. It should be enough to just have the `isEqualTo()` check.